src/main/java/edu/usf/cutr/grha/io/TncToGtfsWriter.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (C) 2021 University of South Florida
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package edu.usf.cutr.grha.io;
+
+import edu.usf.cutr.grha.model.ChicagoTncData;
+import edu.usf.cutr.grha.utils.GtfsUtils;
+import org.onebusaway.gtfs.model.*;
+import org.onebusaway.gtfs.model.calendar.ServiceDate;
+import org.onebusaway.gtfs.serialization.GtfsWriter;
+
+import java.io.File;
+import java.util.Date;
+import java.util.List;
+
+public class TncToGtfsWriter {
+    private static final String FAKE_AGENCY_ID = "chicago-tnc";
+    private static final String FAKE_AGENCY_NAME = "Chicago TNC Trips";
+    private static final String FAKE_AGENCY_URL = "https://data.cityofchicago.org/Transportation/Transportation-Network-Providers-Trips/m6dm-c72p/data";
+    private static final String AGENCY_TIMEZONE = "America/Chicago";
+
+    private static final String FAKE_ROUTE_ID = "1";
+    private static final String FAKE_SHORT_NAME = "TNC Trips";
+
+    private static final String PICKUP_STOP_SUFFIX = "_pickup";
+    private static final String DROPOFF_STOP_SUFFIX = "_dropoff";
+
+    public String filePath;
+
+    public TncToGtfsWriter(String filePath) {
+        this.filePath = filePath;
+    }
+
+    public void write(List<ChicagoTncData> chicagoTncDataList) {
+        GtfsWriter writer = new GtfsWriter();
+        writer.setOutputLocation(new File(filePath));
+
+        // Creates a single fake agency in agency.txt for TNC trips
+        Agency agency = newFakeAgency();
+        writer.handleEntity(agency);
+
+        // Creates a single fake route in routes.txt that all TNC trips can be assigned to
+        Route route = newFakeRoute(agency);
+        writer.handleEntity(route);
+
+        int counter = 1;
+        for (ChicagoTncData tncData: chicagoTncDataList) {
+
+            // Check for data with no location values
+            if (tncData.getPickupCentroidLatitude() == 0 ||
+            tncData.getPickupCentroidLongitude() == 0 ||
+            tncData.getDropoffCentroidLatitude() == 0 ||
+            tncData.getDropoffCentroidLongitude() == 0) {
+                System.out.println("Skipping record due to missing location values: \n" + tncData.toString());
+                counter ++;
```

Comment:
You shouldn't need to increment the `counter` here if you didn't output any stops.

However, could you please add a separate counter here that counts the number of records with missing location values, and print that out at the end as well?